### PR TITLE
adding backend to calendar

### PR DESCRIPTION
**Pull request by jacobayu**
### Administrative Info
Monday Board ID: 853648344
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Connected the backend to the calendar view, so the events in the database now show up in the calendar.

### Testing
How did you confirm your changes worked? 
- Tested calendar on local device. Manually compared database entries to the calendar to make sure the name of the classes and their times were correct.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
[
](url)| [TODO Screenshot Title] |
Screen Shot 2021-04-15 at 9 33 29 PM

| --- |
| [TODO Screenshot Link] |

### CODE REVIEW

**Review by eliasfang — APPROVED** [review]
Nice!

File: app/public/javascripts/CALENDAR.JS [hunk 1]
```diff
@@ -1,49 +1,75 @@
...
+                    events: newEventList,
+                    eventColor: '#FFCB8A',
+                    // eventColor: '#F0AEE1',
+                    // eventColor: '#F2F2B4',
+            
+                    customButtons: {
+                        addEventButton: {
+                            text: 'add event...',
+                            click: function() {
+                                var dateStr = prompt('Enter a date in YYYY-MM-DD format');
```
Comment (mingkaili) [inline app/public/javascripts/CALENDAR.JS]: const instead of var

File: app/public/javascripts/CALENDAR.JS [hunk 2]
```diff
@@ -1,49 +1,75 @@
...
+                                var date = new Date(dateStr + 'T00:00:00'); // will be in local time
```
Comment (mingkaili) [inline app/public/javascripts/CALENDAR.JS]: const instead of var

File: app/public/javascripts/CALENDAR.JS [hunk 3]
```diff
@@ -1,49 +1,75 @@
...
+                    // eventColor: '#F0AEE1',
+                    // eventColor: '#F2F2B4',
+            
+                    customButtons: {
+                        addEventButton: {
+                            text: 'add event...',
+                            click: function() {
+                                var dateStr = prompt('Enter a date in YYYY-MM-DD format');
+                                var date = new Date(dateStr + 'T00:00:00'); // will be in local time          
+                                var titleStr = prompt('Enter a title');
```
Comment (mingkaili) [inline app/public/javascripts/CALENDAR.JS]: const instead of var

File: app/public/javascripts/CALENDAR.JS [hunk 4]
```diff
@@ -1,49 +1,75 @@
...
+                    headerToolbar: {
+                        left: 'prev,next today',
+                        center: 'addEventButton',
+                        // center: 'title',
+                        right: 'timeGridWeek,timeGridDay'
+                    },
+            
+                    editable: true,
+                    dayMaxEvents: true, // when too manyn events in a day, show the popover
+                    //events: 'https://fullcalendar.io/demo-events.json?overload-day',
```
Comment (mingkaili) [inline app/public/javascripts/CALENDAR.JS]: do we need these comments?

File: app/public/javascripts/CALENDAR.JS [hunk 5]
```diff
@@ -1,49 +1,75 @@
...
+                    events: newEventList,
+                    eventColor: '#FFCB8A',
+                    // eventColor: '#F0AEE1',
```
Comment (mingkaili) [inline app/public/javascripts/CALENDAR.JS]: Same here

File: app/public/javascripts/CALENDAR.JS [hunk 6]
```diff
@@ -1,49 +1,75 @@
...
+                    const event = {
+                        title: result[i].class_name,
+                        start: startTimeString,
+                        end: endTimeString
                     }
-
+                    newEventList.push(event);
                 }
-
+                var calendar = new FullCalendar.Calendar(calendarEl, {
```
Comment (mingkaili) [inline app/public/javascripts/CALENDAR.JS]: const

**Review by mingkaili — CHANGES_REQUESTED** [review]

**Review by mingkaili — APPROVED** [review]
LGTM

**Pull request merged**